### PR TITLE
assets: add --by-overall-limit to assets purge command

### LINKED ISSUE #4457: Cache: purge by overall cache size
Besides the existing features, we need to trim the cache to a defined size, so that it matches the storage limitations of users (and CI systems).

For instance it may be the case that a GitLab cache limit is 4 GiB, in that case we can sort by last access, and remove all that exceeds 4 GiB (that is, keep the last accessed 4 GiB worth of cached files).

This would ensure that the cache is automatically being removed of files that were used last (and possibly not used anymore).

This can be available to users as a command such as:

```
avocado assets purge --by-limit=4G
```

This is one last use case of #4311

### PR DESCRIPTION
**Pull request by beraldoleal**
This will remove assets by an overall limit sorting assets by atime.
For instance, it may be the case that a GitLab cache limit is 4 GiB, in
that case, we can sort by last access, and remove all that exceeds 4 GiB
(that is, keep the last accessed 4 GiB worth of cached files). This
fixes #4457.
    
Signed-off-by: Beraldo Leal <bleal@redhat.com>

### CODE REVIEW

File: avocado/plugins/assets.py [hunk 1]
```diff
@@ -268,6 +268,13 @@ class Assets(CLICmd):
     name = 'assets'
     description = 'Manage assets'
 
+    @staticmethod
```
Comment (clebergnu) [inline avocado/plugins/assets.py:272]: Typo in the commit message: s/instad/instead/

File: spell.ignore [hunk 2]
```diff
@@ -741,3 +741,4 @@ crit
 err
 getpcaps
 libcap
+GiB
```
Comment (clebergnu) [inline spell.ignore:744]: For the sake of bisectability, this should be on the commit that introduced it, that is c2fe224c58be812bc799d54e3c6aef7ba77603b3.

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @beraldoleal ,

While running `$ avocado assets purge --by-overall-limit=3GiB` I got:

``` 
Avocado crashed unexpectedly: String not in size + unit format (i.e. "10M", "100k", ...)
You can find details in /home/cleber/avocado/data/crashes/avocado-traceback-2021-04-06_22:36:01-zcxiy9yh.log
```

And its contents:

```
Avocado crashed:
Traceback (most recent call last):
  File "/home/cleber/src/avocado/avocado/avocado/utils/data_structures.py", line 295, in __init__
    self._value = int(norm_size[:-1])
ValueError: invalid literal for int() with base 10: '3gi'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/cleber/.local/bin/avocado", line 33, in <module>
    sys.exit(load_entry_point('avocado-framework', 'console_scripts', 'avocado')())
  File "/home/cleber/src/avocado/avocado/avocado/core/main.py", line 76, in main
    return app.run()
  File "/home/cleber/src/avocado/avocado/avocado/core/app.py", line 112, in run
    return method(self.parser.config)
  File "/home/cleber/src/avocado/avocado/avocado/plugins/assets.py", line 516, in run
    return self.handle_purge(config)
  File "/home/cleber/src/avocado/avocado/avocado/plugins/assets.py", line 413, in handle_purge
    size = DataSize(overall_limit).b
  File "/home/cleber/src/avocado/avocado/avocado/utils/data_structures.py", line 304, in __init__
    raise InvalidDataSize('String not in size + unit format (i.e. '
avocado.utils.data_structures.InvalidDataSize: String not in size + unit format (i.e. "10M", "100k", ...)
```

I tried with that unit, because it's mentioned in the docstrings.  Also, in the option help message, you should try to get the valid suffixes from `DataSize.MULTIPLIERS`.

Other than that, the feature seems to be working.  I had about 3.5 Gb, and asked it to reduce to 3.0 Gb, and the different between before and after was:

```diff
--- /tmp/pre    2021-04-06 22:35:20.793349859 -0400
+++ /tmp/post   2021-04-06 22:39:04.749585813 -0400
@@ -68,16 +68,3 @@
 Armbian_20.08.1_Orangepipc_bionic_current_5.8.5.img.xz       31fff0c268 2021-02-15 22:36:44 734.00 B
 u-boot-sunxi_2020.01%2Bdfsg-1_armhf.deb                      f67f404a80 2021-02-11 19:28:45 1.46 MB
 Armbian_20.08.1_Orangepipc_bionic_current_5.8.5.img.xz       b4d6775f56 2021-02-11 19:28:43 287.50 MB
-Armbian_19.11.3_Orangepipc_bionic_current_5.3.9.7z           196a8ffb72 2021-02-11 19:26:54 205.16 MB
-rootfs.ext2.xz                                               692510cb62 2021-02-11 19:24:57 2.90 MB
-linux-image-dev-sunxi_5.75_armhf.deb                         1334c29c44 2021-02-11 19:24:54 20.33 MB
-linux-image-dev-sunxi_5.75_armhf.deb                         1334c29c44 2021-02-11 19:24:45 20.33 MB
-generic_com0.fs                                              dbcfc09912 2021-02-11 19:09:27 1.90 MB
-P12H0456.IMG                                                 1775face4e 2021-02-11 19:09:26 524.29 KB
-HelenOS-0.6.0-sparc32-leon3.bin                              a88c9cfdb8 2021-02-11 19:09:25 2.42 MB
-logo_linux_vga16.ppm                                         3991c2ddbd 2021-02-11 19:08:13 76.84 KB
-armv7.img.gz                                                 2babb29d36 2021-02-11 19:08:01 317.74 MB
-u-boot-sunxi_2020.01%2Bdfsg-1_armhf.deb                      f67f404a80 2021-02-11 19:03:38 1.46 MB
-obmc-phosphor-image-gsj.static.mtd.gz                        14895e6349 2021-02-11 19:03:26 26.60 MB
-rootfs.mipsel64r1.cpio.gz                                    1dbb8a396e 2021-02-11 19:02:45 2.04 MB
-vmlinux-3.19.3.mtoman.20150408                               00d1d268fb 2021-02-11 19:02:45 12.90 MB
```

Which resulted in:

```
$ du -hcs ~/avocado/data/cache/
3.0G    /home/cleber/avocado/data/cache/
3.0G    total
```

**Comment by beraldoleal** [comment]
Closing in favor of a v2.

**Pull request closed without merging**